python_modules/libraries/dagster-dbt/dagster_dbt/cloud_v2/cli_invocation.py

```diff
@@ -1,48 +1,44 @@
-from collections.abc import Iterator, Sequence
-from typing import TYPE_CHECKING, Union
+from collections.abc import Iterator, Mapping, Sequence
+from typing import Any, Union
 
 from dagster import AssetCheckEvaluation, AssetMaterialization
 from dagster._annotations import preview
+from dagster._record import record
 
+from dagster_dbt.cloud_v2.client import DbtCloudWorkspaceClient
 from dagster_dbt.cloud_v2.run_handler import DbtCloudJobRunHandler, DbtCloudJobRunResults
 from dagster_dbt.dagster_dbt_translator import DagsterDbtTranslator
 
-if TYPE_CHECKING:
-    from dagster_dbt.cloud_v2.resources import DbtCloudWorkspace
-
 
 @preview
+@record
 class DbtCloudCliInvocation:
     """Represents a dbt Cloud cli invocation."""
 
-    def __init__(
-        self,
-        args: Sequence[str],
-        workspace: "DbtCloudWorkspace",
-        dagster_dbt_translator: DagsterDbtTranslator,
-        run_handler: DbtCloudJobRunHandler,
-    ):
-        self.args = args
-        self.workspace = workspace
-        self.dagster_dbt_translator = dagster_dbt_translator
-        self.run_handler = run_handler
+    args: Sequence[str]
+    client: DbtCloudWorkspaceClient
+    manifest: Mapping[str, Any]
```

Comment:
I don't understand; when would this ever be materially different than `workspace.manifest?`. Is the idea that it's being generated by the run?

Reply:
When we create the asset spec, we attach the manifest and translator to its metadata, to make sure that the exact same are used at materialization time. In principle, the workspace.manifest should be the same as the manifest stored in the asset defs, but this is mainly to avoid conflicts. 

This behavior is also implemented in dbt core.

Reply:
Okay gotcha; just leave a comment